### ISSUE TITLE
Memory leaks in redirects and queues objects

### ISSUE THREAD

**Issue by markbrown4** [bradoyler/route-cache#24]
The redirects object is never emptied so keeps on growing as new redirects are cached.
The redirects also never expire which I'd expect from a method named `cacheSeconds()`

You end up with an object like this that keeps growing:

```js
redirects = {
  "url1": { url: "url1a" , status: 301 },
  "url2": { url: "url2a" , status: 301 },
  "url3": { url: "url3a" , status: 301 }
}
```

Perhaps a fix would be putting the redirects in the lru-cache too so they hit a limit / expire natually.

**Comment by markbrown4** [comment]
Actually, the queues object appears leaky also..

The items are cleared out but the keys persist so you end up with an object like this that never stops growing

```
queues = {
  "url1": [],
  "url2": [],
  "url3": []
}
```

**Comment by markbrown4** [comment]
Can see an implementation here of the above:

https://github.com/markbrown4/route-cache/commit/162efc66218061da99ec49b7bbaaadbfed6d35aa

https://github.com/markbrown4/route-cache/commit/dc34e8eb3c00c73fed4c70bf9572e57df797e1af

**Comment by bradoyler** [comment]
@markbrown4 awesome. Nice work.

**Comment by bradoyler** [comment]
@markbrown4 u have a PR for this or want to address #25 first?

**Comment by markbrown4** [comment]
This is the more important issue so we'll solve this one first.

I'll get a PR together.  It will be close to what I have above but I'll need to store status codes for the redirects too as you have tests around 301's and 302's.

**Comment by bradoyler** [comment]
Would u be able to repro the issue with a mocha test?
I still haven't been able to reproduce the issue.

**Comment by markbrown4** [comment]
I can't write a test for the leak without changing the code to expose these objects:

```js
var queues = {}
var redirects = {}
```

You're adding keys to these objects but never removing them so as the number of keys increases so does the size of these objects.

I could add a failing test to confirm the redirects don't expire in the ttl though.

**Comment by bradoyler** [comment]
Closing as its is mostly resolved with #26 
I'll look into the `{queues}` memory consumption next. Feel free to open an issue if ya like.
